src/HeaderSort/components/HeaderSort.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, createElement } from "react";
+import * as classNames from "classnames";
+
+export interface HeaderSortProps {
+    caption: string;
+    friendlyId?: string;
+    initialSorted: boolean;
+    onClickAction?: (attribute: string, order: string) => void;
+    publishedSortAttribute?: string;
+    publishedSortOrder?: SortOrder;
+    publishedSortWidgetFriendlyId?: string;
+    sortAttribute: string;
+    sortOrder: SortOrder;
+}
+
+export type SortOrder = "" | "desc" | "asc";
```

Comment:
Would it be better to have explicit name no sorting, `"none" | "desc" | "asc";`
We could also create extemde type, so `SortOrder` type could be re-used in the `ContainerProps`
``` ts
export type SortOrder = "desc" | "asc";
export type SortOrderState = SortOrder | "none";
```